### PR TITLE
Order polytope related fixes and improvements

### PR DESCRIPTION
**Pull request by vaithak**
The changes in the PR have been pulled apart from #171 for easy reviewal and better readability.

### CODE REVIEW

File: include/convex_bodies/orderpolytope.h [hunk 1]
```diff
@@ -216,21 +208,16 @@ class OrderPolytope {
...
-
-        // change entries of A, doing here as won't be required in
-        // optimized volume calculation of order-polytope
-        MT A = _A.rowwise().normalized();
+        normalize();
         std::pair<Point, NT> _innerball;
 
-        #ifndef VOLESTIPY   // as _A is never normalized in closed form
-            _innerball = ComputeChebychevBall<NT, Point>(A, b); // use lpsolve library
+        #ifndef VOLESTIPY
```
Comment (TolisChal) [inline include/convex_bodies/orderpolytope.h]: VOLESTIPY is always undefined. Plz use only `ComputeChebychevBall<NT, Point>(A, b)`

File: include/generators/order_polytope_generator.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// generates a cube as an Order Polytope
+template <class Polytope>
+Polytope generate_cube_orderpoly(unsigned int dim) {
+    typedef typename Poset::RV RV;
+
+    RV order_relations;
+    Poset poset(dim, order_relations);
+    Polytope OP(poset);
+    return OP;
+}
```
Comment (TolisChal) [inline include/generators/order_polytope_generator.h]: Plz add a blank line

File: include/convex_bodies/orderpolytope.h (hunk 1 above)
Comment (vissarion) [reply include/convex_bodies/orderpolytope.h]: @TolisChal  is VOLESTIPY needed in the library now?

File: include/convex_bodies/orderpolytope.h [hunk 3]
```diff
@@ -188,11 +180,11 @@ class OrderPolytope {
             // DON'T JUST check violation of point between 0 and 1
             // as b will change for shifted polytope
             diff = -pt_coeffs(i) - b(i);
-            if (_normalized)    diff /= _row_norms(i);
+            // if (_normalized)    diff /= _row_norms(i); // row_norm is 1
```
Comment (vissarion) [inline include/convex_bodies/orderpolytope.h]: is this needed as a comment? If not please delete.

File: include/convex_bodies/orderpolytope.h [hunk 4]
```diff
@@ -188,11 +180,11 @@ class OrderPolytope {
...
             if (diff > NT(tol)) return 0;
 
             diff = pt_coeffs(i) - b(i + _d);
-            if (_normalized)    diff /= _row_norms(i + _d);
+            // if (_normalized)    diff /= _row_norms(i + _d);  // row_norm is 1
```
Comment (vissarion) [inline include/convex_bodies/orderpolytope.h]: same here

File: include/generators/order_polytope_generator.h [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                                       0 1 0 0 0 0 0 0
+                                       0 0 0 0 0 0 0 0
+                                       0 0 1 1 0 0 0 0
+                                       0 0 0 0 1 0 0 0
+                                       0 0 0 0 0 0 0 0
+                                       0 0 0 0 1 1 1 0)"},
+
+};
+
+// generates an Order Polytope from an instance file containing the adjacency matrix
```
Comment (vissarion) [inline include/generators/order_polytope_generator.h]: In the comment you say that the function takes a file but it takes a string, am I missing something?

**Review by vissarion — APPROVED** [review]
Thanks! I have a few comments, otherwise OK for merging.

File: include/convex_bodies/orderpolytope.h (hunk 1 above)
Comment (vissarion) [reply include/convex_bodies/orderpolytope.h]: https://github.com/GeomScale/volume_approximation/issues/148

File: include/generators/order_polytope_generator.h (hunk 5 above)
Comment (vaithak) [reply include/generators/order_polytope_generator.h]: Initially, my idea was to have a directory containing all the instances as txt files, but later changed to using an unordered_map. Fixed the comment 👍🏽

File: include/convex_bodies/orderpolytope.h (hunk 4 above)
Comment (vaithak) [reply include/convex_bodies/orderpolytope.h]: removed 👍🏽

File: include/generators/order_polytope_generator.h (hunk 2 above)
Comment (vaithak) [reply include/generators/order_polytope_generator.h]: added 👍🏽

File: include/convex_bodies/orderpolytope.h (hunk 3 above)
Comment (vaithak) [reply include/convex_bodies/orderpolytope.h]: removed 👍🏽

File: include/convex_bodies/orderpolytope.h (hunk 1 above)
Comment (vaithak) [reply include/convex_bodies/orderpolytope.h]: now the function returns just `ComputeChebychevBall<NT, Point>(A, b)`

**Review by TolisChal — APPROVED** [review]

**Pull request merged**